app/models/developers/external_profiles/linkedin_profile_fetcher.rb

```diff
@@ -1,40 +1,32 @@
 module Developers::ExternalProfiles
   class LinkedinProfileFetcher
     def developer_profiles
-      profiles = []
       developers = developers_with_linkedin_profiles
       Rails.logger.info "Fetching #{developers.count} #{"LinkedIn profile".pluralize(developers.count)}..."
 
       developers.find_each.with_index do |developer, index|
         response = fetch_linkedin_profile(developer.linkedin)
         record = external_profile(developer, response)
-        profiles << record if record.present?
+        Developers::ExternalProfile.upsert(record, unique_by: [:developer_id, :site]) if record.present?
```

Comment:
Using upsert for a single record doesn't feel right to me. I think we want to instead use something like `find_or_initialize_by` or `find_or_create_by`. Then we can simplify this logic to find/initialize by the developer/site and update only if necessary.